xcc/settings.py

```diff
@@ -87,14 +94,42 @@ class Config:  # pylint: disable=missing-class-docstring
         env_file = get_path_to_env_file()
         env_prefix = get_name_of_env_var()
 
+    def _check_for_invalid_values(self, key: str, val: str) -> None:
+        """
+        Check for conditions that make saving the env_file
+        dangerous to the user.
+
+        - REFRESH_TOKEN must not contain characters outside Base64URL set
+
+        Args:
+            key (str): .env file key
+            val (str): .env file value
+
+        Raises:
+            Exception: if the value should not be saved to the .env file
+
+        """
+
+        if (
+            key == "REFRESH_TOKEN"
+            and val is not None
+            and re.search(_BASE64URLRE, val) is not None
```

Comment:
Hmm, why is this checking for `not None` rather than `None`? 🤔 I thought the idea is that we want to raise an exception if the regular expression is _not_ matched:
```suggestion
            and re.search(_BASE64URLRE, val) is None
```

Reply:
We want it to be `None`. If it `is not None` that means we found a forbidden character.

Reply:
Ah, sorry, you're right. I missed the fact the regex was matching characters _not_ in the set of JWT characters.